pkg/kube/client.go

```diff
@@ -479,14 +479,21 @@ func newClientInternal(clientFactory *clientFactory, opts ...ClientOption) (*cli
 		return nil, err
 	}
 
-	c.http = &http.Client{
-		Timeout: time.Second * 15,
+	c.http = &http.Client{}
+	if c.config != nil && c.config.Timeout != 0 {
+		c.http.Timeout = c.config.Timeout
+	} else {
+		c.http.Timeout = time.Second * 5
```

Comment:
If the caller doesn't explicitly configure the timeout, we're changing the default from 15 seconds to 5 seconds. Was this intentional or a typo?

Reply:
Typo :man_facepalming: